src/NTH.Windows.Forms/NTH.Windows.Forms/NthListView.cs

```diff
@@ -1,48 +1,111 @@
+        //        {
+        //            HWnd = Handle,
+        //            Msg = WM_PRINTCLIENT,
+        //            WParam = e.Graphics.GetHdc(),
+        //            LParam = (IntPtr) PRF_CLIENT
+        //        };
+        //        DefWndProc(ref m);
+        //        e.Graphics.ReleaseHdc(m.WParam);
+        //    }
+        //    base.OnPaint(e);
+        //}
+
+        public void MakeCollapsable() // Adds expanders to the groups, should be called in form's Show-event
         {
-            SetStyle(ControlStyles.OptimizedDoubleBuffer | ControlStyles.AllPaintingInWmPaint, true);
-            // Disable default CommCtrl painting on non-Vista systems
             if (Environment.OSVersion.Version.Major < 6)
-                SetStyle(ControlStyles.UserPaint, true);
+                return;
+
+            const int lvmFirst = 0x1000;
+            const int LVM_SETGROUPINFO = (lvmFirst + 147);
+
+            foreach (ListViewGroup group in Groups)
+            {
+                var placeHolderGroup = new NthListViewGroup();
+                placeHolderGroup.CbSize = Marshal.SizeOf(placeHolderGroup);
+                placeHolderGroup.State = NthListViewGroupState.Collapsible;
+                placeHolderGroup.Mask = NthListViewGroupMask.State;
+                placeHolderGroup.GroupId = GetGroupId(group);
+
+                if (placeHolderGroup.GroupId >= 0)
+                    NativeMethods.SendMessage(Handle, LVM_SETGROUPINFO, new IntPtr(placeHolderGroup.GroupId), ref placeHolderGroup);
+            }
         }
 
-        protected override void OnPaint(PaintEventArgs e)
+        private static int GetGroupId(ListViewGroup group)
         {
-            if (GetStyle(ControlStyles.UserPaint))
+            var groupType = group.GetType();
             {
-                Message m = new Message();
-                m.HWnd = Handle;
-                m.Msg = WM_PRINTCLIENT;
-                m.WParam = e.Graphics.GetHdc();
-                m.LParam = (IntPtr)PRF_CLIENT;
+                var groupIdProperty = groupType.GetProperty("ID", BindingFlags.NonPublic | BindingFlags.Instance); // Include inner fields and instance members
+                if (groupIdProperty == null)
+                    return -1;
+                var value = groupIdProperty.GetValue(group, null);
+                if (value != null)
+                    return (int)value;
+            }
+            return -1;
+        }
+
+        protected override void WndProc(ref Message m)
+        {
+            const int WM_LBUTTONUP = 0x202;
+            if (m.Msg == WM_LBUTTONUP && Environment.OSVersion.Version.Major >= 6)
                 DefWndProc(ref m);
-                e.Graphics.ReleaseHdc(m.WParam);
+
+            switch (m.Msg)
+            {
+                case 15:
+                    if (!_isExplorerListView)
+                    {
+                        NativeMethods.SetWindowTheme(Handle, "explorer", null);
```

Comment:
SetWindowTheme seems to be supported since XP:
https://msdn.microsoft.com/en-us/library/windows/desktop/bb759827%28v=vs.85%29.aspx
So this should be fine.